File: client/android/app/src/main/java/org/cna/keyple/demo/sale/android/nfc/slave/data/model/DeviceEnum.java
/********************************************************************************
 * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
 ********************************************************************************/
package org.cna.keyple.demo.sale.android.nfc.slave.data.model;

import org.cna.keyple.demo.sale.android.nfc.slave.R;

public enum DeviceEnum {
    CONTACTLESS_CARD(R.string.contactless_card), SIM(R.string.sim_card), BRACELET(
            R.string.bracelet), MULTI_CARD(R.string.multi_service_card);

    private final int stringId;

    /**
     * @param stringId
     */
    DeviceEnum(final int stringId) {
        this.stringId = stringId;
    }

    public int getTextId() {
        return stringId;
    }

    public static DeviceEnum getDeviceEnum(String name) {
        try {
            return DeviceEnum.valueOf(name.toUpperCase());
        } catch (Exception e) {
            // If the given state does not exist, return the default value.
            return DeviceEnum.CONTACTLESS_CARD;
        }
    }
}

File: client/android/app/src/main/kotlin/org/eclipse/keyple/demo/remote/data/model/DeviceEnum.kt
/*
 * Copyright (c) 2020 Calypso Networks Association https://www.calypsonet-asso.org/
 *
 * See the NOTICE file(s) distributed with this work for additional information
 * regarding copyright ownership.
 *
 * This program and the accompanying materials are made available under the terms of the
 * Eclipse Public License 2.0 which is available at http://www.eclipse.org/legal/epl-2.0
 *
 * SPDX-License-Identifier: EPL-2.0
 */
package org.eclipse.keyple.demo.remote.data.model

import org.eclipse.keyple.demo.remote.R

enum class DeviceEnum
/**
 * @param stringId
 */(val textId: Int) {
    CONTACTLESS_CARD(R.string.contactless_card), SIM(R.string.sim_card), WEARABLE(
        R.string.wearable
    ),
    EMBEDDED(R.string.embedded_secure_elem);

    companion object {
        @JvmStatic
        fun getDeviceEnum(name: String): DeviceEnum {
            return try {
                valueOf(name.toUpperCase())
            } catch (e: Exception) {
                // If the given state does not exist, return the default value.
                CONTACTLESS_CARD
            }
        }
    }
}